### PR TITLE
Use `--nworkers` in stead of deprecated `--nprocs` in the generated job scripts

### PR DESCRIPTION
**Pull request by ikabadzhov**
Fix https://github.com/dask/dask-jobqueue/issues/559.

### CODE REVIEW

**Review by guillaumeeb — COMMENTED** [review]
Thanks @ikabadzhov.

Unfortunately, test will fail as CI is borken, but this is not a problem to merge. You didn't let any nprocs in the code?

Anyway, I'm thinking about one thing: we should probably update the minimum required version of Dask/Distributed. Did you check when `nworkers` was introduced?

**Comment by ikabadzhov** [comment]
I did not leave any `--nprocs` that is right.

I think the deprecation was introduced here https://github.com/dask/distributed/blame/main/distributed/cli/dask_ssh.py#L52 (6 months ago). I am not sure how to map it to the right version.

File: dask_jobqueue/tests/test_job.py
Comment (ikabadzhov) [inline dask_jobqueue/tests/test_job.py:116]: I also changed the name of this test. That might make the CI unhappy.

**Comment by guillaumeeb** [comment]
Ok, so the new option name was introduced in 2022.02.0, see https://distributed.dask.org/en/stable/changelog.html#v2022-02-0.

So you should also update required distributed version with this one.

As we just released a dask-jobqueue version, I belive it is safe to just do that.

File: dask_jobqueue/tests/test_job.py
Comment (guillaumeeb) [reply dask_jobqueue/tests/test_job.py:116]: CI problems are unfortunately a bit old. I'm hopping to have a look at it in the following weeks.

**Comment by guillaumeeb** [comment]
Also, There is on failure related to your PR in the CI: the check of formatting using black. Could you run black on your branch to format things correctly?

**Comment by ikabadzhov** [comment]
I just applied `black`. I am not sure for this repository to have code that is not formatted, so I kept all changes in a separate commit. Let me know if I should format only my changes.

**Comment by ikabadzhov** [comment]
> Ok, so the new option name was introduced in 2022.02.0, see [distributed.dask.org/en/stable/changelog.html#v2022-02-0](https://distributed.dask.org/en/stable/changelog.html#v2022-02-0).
> 
> So you should also update required distributed version with this one.
> 
> As we just released a dask-jobqueue version, I belive it is safe to just do that.

I just updated the distributed requirement, now it is:
```py
dask>=2.23
distributed>=2022.02.0
```
Is this what you mean? Shouldn't they be the same?

**Comment by guillaumeeb** [comment]
> Is this what you mean? Shouldn't they be the same?

This is what I mean, but yes I think they should be the same.

We may also need to wait for #562 to be in to merge this one, but not sure about it.

**Review by jacobtomlinson — COMMENTED** [review]
Since switching to CalVer we release `dask` and `distributed` in lock step and only guarantee that those releases work together. Generally pinning will mean that installing a more recent version of `distributed` will upgrade `dask` to the same version, but to be explicit we might want to bump the minimum `dask` version here to match `distributed`.

**Comment by jacobtomlinson** [comment]
_We squash and merge here so please don't squash and force push prematurely, it makes review harder as it's unclear what has changed since the last review._

**Review by guillaumeeb — APPROVED** [review]
Anyway, I checked for nprocs occurences, and all are well taken into account. The requirement version of Dask and distributed are OK too, this is good for me.

I still want to advance a bit on #562 before merging.

@jacobtomlinson any other comment.

**Comment by jacobtomlinson** [comment]
LGTM

**Comment by guillaumeeb** [comment]
All green, merging!

**Pull request merged**
